client/app/scripts/liveblog-edit/directives.js

```diff
@@ -50,40 +50,116 @@ define([
                 }
             };
         })
-        .directive('draftPosts', ['api', 'postsService',
-            function DraftPostsDirective(api, postsService) {
-                DraftPostsController.$inject = ['$scope'];
-                function DraftPostsController($scope) {
-                    var mv = this;
-                    mv.posts = [];
-                    mv.postsInfo = {};
-                    mv.selectDraftPost = $scope.onDraftPostSeleted;
-                    function updateList() {
-                        postsService.fetchDrafts(
-                            $scope.blog._id,
-                            mv.posts,
-                            mv.postsInfo
-                        );
+        .directive('lbPostsList', [
+            'postsService', 'notify', '$q',
+            function(postsService, notify, $q) {
+
+                LbPostsListCtrl.$inject = ['$scope'];
+                function LbPostsListCtrl($scope) {
+                    var vm = this;
+
+                    function fetchPage() {
+                        // stop if the limit is reached
+                        if (angular.isDefined(vm.postsMeta.total) && vm.postsMeta.total <= vm.posts.length) {
+                            return false;
+                        }
+                        // Find the next page containing new posts
+                        var page = 1;
+                        while (vm.pagination.limit * page <= vm.posts.length) {
```

Comment:
For me, here this `while` is more explicit and natural than an euclidean division.
It's because I'm more a developer than a mathematician :), and given the low number of pages expected, it won't cost too much to the CPU.
Why it shouldn't work ? Can you explain the problem we will have ?
As it's done, for me, we will retrieve every time the next page containing new post. What ever the number of posts in the list.
For the updates, the mechanism is bellow.